third_party/proxy/service.go

```diff
@@ -230,7 +230,13 @@ func (sct *ServiceChangeTracker) newBaseServiceInfo(port *v1.ServicePort, servic
 		externalPolicyLocal:   externalPolicyLocal,
 		internalPolicyLocal:   internalPolicyLocal,
 		internalTrafficPolicy: service.Spec.InternalTrafficPolicy,
-		hintsAnnotation:       service.Annotations[v1.AnnotationTopologyAwareHints],
+	}
+	// TODO: Switch to v1.DeprecatedAnnotationTopologyAwareHints and v1.AnnotationTopologyMode after
+	//  upgrading Antrea K8s API to at least 1.27
+	var ok bool
+	info.hintsAnnotation, ok = service.Annotations[v1.AnnotationTopologyAwareHints]
+	if !ok {
+		info.hintsAnnotation, _ = service.Annotations["service.kubernetes.io/topology-mode"]
```

Comment:
I think the new annotation should be preferred when both annotations exist.

Reply:
Ignore the comment, I see you are following the same behavior as kube-proxy.

Reply:
I checked K8s implementation, v1.DeprecatedAnnotationTopologyAwareHints has precedence over v1.AnnotationTopologyMode in 1.27. https://github.com/kubernetes/kubernetes/blob/v1.27.3/pkg/proxy/service.go#L181